tests/MongoDB.Driver.Tests/Linq/MongoQueryableTests.cs

```diff
@@ -1136,6 +1134,158 @@ public void Select_followed_by_group()
+            commonResult[1].Result.Count().Should().Be(2);
+            commonResult[1].Result.First().N.Should().Be(3);
+            commonResult[1].Result.Last().N.Should().Be(4);
+
+            query = CreateQuery().Select(g => new { Result = g.G.Where(g1 => g1.N > (g.G.Count(g2 => g2.N > g.C.E.F) + g.C.N)) });
+
+            commonResult = Assert(
+                query,
+                2,
+                "{ $project:{ 'Result' : { '$filter' : { 'input' : '$G', 'as' : 'g1', 'cond' : { '$gt' : ['$$g1.N', { '$add' : [{ '$size' : { '$filter' : { 'input' : '$G', 'as' : 'g2', 'cond' : { '$gt' : ['$$g2.N', '$C.E.F'] } } } }, '$C.N'] }] } } }, '_id' : 0 } }");
+
+            commonResult[0].Result.Count().Should().Be(2);
+            commonResult[0].Result.First().N.Should().Be(1);
+            commonResult[0].Result.Last().N.Should().Be(2);
+            commonResult[1].Result.Count().Should().Be(2);
+            commonResult[1].Result.First().N.Should().Be(3);
+            commonResult[1].Result.Last().N.Should().Be(4);
+        }
+
+        [SkippableFact]
+        public void Select_with_using_top_level_expression_input_parameters_inside_children_levels_and_with_inner_select()
+        {
+            RequireServer.Check().Supports(Feature.AggregateArrayFilter);
+
+            var query = CreateQuery()
+                .Select(g => new { Result = g.G.Where(g1 => g1.N > g.G.Select(g2 => g2.E).Min(g3 => g3.F)) });
+
+            var commonResult = Assert(
+                query,
+                2,
+                "{ '$project' : { 'Result' : { '$filter' : { 'input' : '$G', 'as' : 'g1', 'cond' : { '$gt' : ['$$g1.N', { '$min' : '$G.E.F' }] } } }, '_id' : 0 } }");
+
+            commonResult[0].Result.Count().Should().Be(0);
+            commonResult[1].Result.Count().Should().Be(0);
+        }
+
+        [SkippableFact]
+        public void Select_with_using_local_collection_inside_children_expression_levels()
+        {
+            RequireServer.Check().Supports(Feature.AggregateArrayFilter);
+
+            var local = new[] { 1, 2, 3 };
+            var query = CreateQuery()
+                .Select(g => new { Result = local.Where(g1 => g1 == g.G.Min(g2 => g2.N)) });
+
+            var commonResult = Assert(
+                query,
+                2,
+                "{ '$project' : { 'Result' : { '$filter' : { 'input' : [1, 2, 3], 'as' : 'g1', 'cond' : { '$eq' : ['$$g1', { '$min' : '$G.N' }] } } }, '_id' : 0 } }");
+
+            commonResult[0].Result.Count().Should().Be(1);
+            commonResult[0].Result.First().Should().Be(1);
+            commonResult[1].Result.Count().Should().Be(1);
+            commonResult[1].Result.First().Should().Be(3);
+        }
+
+        //[SkippableFact]
+        //public void Select_with_using_top_level_expression_input_parameters_inside_children_levels_and_with_inner_select22()
+        //{
+        //this test doesn't return correct Select result, but it looks like it's minor problem.
```

Comment:
Is this in scope to solve here? The title of 2269 seems to indicate it is...

Reply:
Good point. I wasn't sure that we should care about this case in the scope of this ticket.
This test is about having an out of scope expression parameter in `Select` predicate which doesn't look like often situation. But I've fixed this case too. See my last commit.

I also had some other tests which were not included to the final commit since they had Linq expressions which are not so close to this ticket description. I guess I will create tests for them and then we will decide should we worry about them now or not. Maybe I will create new tickets for them.